skills/csharp/experimental/itsmskill/Bots/DefaultActivityHandler.cs

```diff
@@ -53,14 +67,91 @@ protected override Task OnTeamsSigninVerifyStateAsync(ITurnContext<IInvokeActivi
             return _dialog.RunAsync(turnContext, _dialogStateAccessor, cancellationToken);
         }
 
-        protected override Task OnEventActivityAsync(ITurnContext<IEventActivity> turnContext, CancellationToken cancellationToken)
+        protected override async Task OnEventActivityAsync(ITurnContext<IEventActivity> turnContext, CancellationToken cancellationToken)
         {
-            return _dialog.RunAsync(turnContext, _dialogStateAccessor, cancellationToken);
+            var ev = turnContext.Activity.AsEventActivity();
+            var value = ev.Value?.ToString();
+
+            switch (ev.Name)
+            {
+                case TokenEvents.TokenResponseEventName:
+                    {
+                        // Forward the token response activity to the dialog waiting on the stack.
+                        await _dialog.RunAsync(turnContext, _dialogStateAccessor, cancellationToken);
+                        break;
+                    }
+
+                case ServiceNowEvents.Proactive:
+                    {
+                        var eventData = JsonConvert.DeserializeObject<ServiceNowNotification>(turnContext.Activity.Value.ToString());
+
+                        var proactiveModel = await _proactiveStateAccessor.GetAsync(turnContext, () => new ProactiveModel());
+
+                        // TODO: Implement a proactive subscription manager for mapping Notification to ConversationReference
+                        var conversationReference = proactiveModel["Key"].Conversation;
+
+                        await turnContext.Adapter.ContinueConversationAsync(_appCredentials.MicrosoftAppId, conversationReference, ContinueConversationCallback(turnContext, eventData), cancellationToken);
+                        break;
+                    }
+
+                default:
+                    {
+                        await turnContext.SendActivityAsync(new Activity(type: ActivityTypes.Trace, text: $"Unknown Event '{ev.Name ?? "undefined"}' was received but not processed."));
```

Comment:
We should always call ` _dialog.RunAsync(turnContext, _dialogStateAccessor, cancellationToken);` for other events.
No need to handle them (like TokenEvents.TokenResponseEventName) here.

Reply:
@xieofxie thanks added that in default